tree.go

```diff
@@ -915,12 +935,16 @@ func (n *InternalNode) GetProofItems(keys keylist, resolver NodeResolverFn) (*Pr
 		// commitment, as the value is 0.
 		_, isempty := n.children[childIdx].(Empty)
 		if isempty {
-			// A question arises here: what if this proof of absence
-			// corresponds to several stems? Should the ext status be
-			// repeated as many times? It would be wasteful, so the
-			// decoding code has to be aware of this corner case.
-			esses = append(esses, extStatusAbsentEmpty|((n.depth+1)<<3))
+			addedStems := map[string]struct{}{}
 			for i := 0; i < len(group); i++ {
+				if _, ok := addedStems[string(group[i][:StemSize])]; !ok {
+					// A question arises here: what if this proof of absence
+					// corresponds to several stems? Should the ext status be
+					// repeated as many times? It's wasteful, so consider if the
+					// decoding code can be aware of this corner case.
+					esses = append(esses, extStatusAbsentEmpty|((n.depth+1)<<3))
+					addedStems[string(group[i][:StemSize])] = struct{}{}
+				}
```

Comment:
I understand the point of the comment, but unfortunately it isn't that easy to do this without buying again all the corner cases.

I think the comment is still a fair take on the situation, so that's why I'm leaving it here, but if somebody wants to propose doing that optimization, it should:
- Do the optimization
- Add tests for that optimization
- Run all Kaustinen to triple-check stuff

Of course, the complicated point here is to asses if the amount of bytes we'd save in a real chain is meaningful for the complexity we'd be adding. (Since this is proposing breaking the 1:1 between stems and extension statuses).